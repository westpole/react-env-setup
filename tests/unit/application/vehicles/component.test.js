import toJson from 'enzyme-to-json';

import Component from '../../../../src/scripts/application/vehicles/component.jsx';
import { mockComponent } from '../../../tools/mock-component.jsx';

describe('Application/Vehicles/Component', () => {
  const spy = jest.fn();

  it('should match saved snapshot', () => {
    const wrapper = mockComponent({
      type: 'shallow',
      scope: Component,
      props: {
        fetchVehicles: spy,
      },
    });

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should fetch data after load', () => {
    const spyComponentDidMount = jest.spyOn(Component.prototype, 'componentDidMount');

    const wrapper = mockComponent({
      type: 'shallow',
      scope: Component,
      props: {
        fetchVehicles: spy,
      },
    });

    expect(wrapper.find('[fetchVehicles]').length).toEqual(1);
    expect(spyComponentDidMount.mock.calls.length).toEqual(1);
  });
});
