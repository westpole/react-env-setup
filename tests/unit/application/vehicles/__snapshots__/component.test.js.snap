// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application/Vehicles/Component should match saved snapshot 1`] = `
<article
  className="vehicles-wrapper"
>
  <h1>
    Star Wars vehicles dictionary
  </h1>
  <Connect(ListComponent)
    fetchVehicles={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />
</article>
`;
